### PR TITLE
feat(statics): add expiring tx CoinFeature

### PR DESCRIPTION
**Pull request by jayfgrollman**
Adds a CoinFeature called expiring_txs to statics for coins like DOT

CLOSES TICKET: BG-67958

We currently have coin specific logic in the trust admin tool, which is a no no. The logic involves DOT, which has expiring transaction times. Instead of using coin specific logic, the best path forward is to add a coinfeature in statics, especially because there may be more coins like this in the future. This PR adds an expiring txs coinfeature, and adds that feature to DOT.

### CODE REVIEW

File: modules/statics/src/base.ts [hunk 1]
```diff
@@ -181,6 +181,10 @@ export enum CoinFeature {
    * This coin supports custody in BitGo Switzerland entities
    */
   CUSTODY_BITGO_SWITZERLAND = 'custody-bitgo-switzerland',
+  /*
+   * This coin has transactions that expire after a certain amount of time.
+   */
+  EXPIRING_TXS = 'expiring-txs',
```
Comment (sachushaji) [inline modules/statics/src/base.ts:187]:
```suggestion
  EXPIRING_TRANSACTIONS = 'expiring-transactions',
```
Lets avoid shorthands for a coin feature

File: modules/statics/src/coins.ts [hunk 2]
```diff
@@ -101,7 +101,7 @@ const ALGO_FEATURES = [
   CoinFeature.CUSTODY_BITGO_SWITZERLAND,
   CoinFeature.CUSTODY_BITGO_GERMANY,
 ];
-const DOT_FEATURES = [...AccountCoin.DEFAULT_FEATURES, CoinFeature.TSS, CoinFeature.STAKING];
+const DOT_FEATURES = [...AccountCoin.DEFAULT_FEATURES, CoinFeature.TSS, CoinFeature.STAKING, CoinFeature.EXPIRING_TXS];
```
Comment (sachushaji) [inline modules/statics/src/coins.ts:104]:
```suggestion
const DOT_FEATURES = [...AccountCoin.DEFAULT_FEATURES, CoinFeature.TSS, CoinFeature.STAKING, CoinFeature. EXPIRING_TRANSACTIONS];
```

File: modules/statics/src/base.ts (hunk 1 above)
Comment (jayfgrollman) [reply modules/statics/src/base.ts]: Done!

File: modules/statics/src/coins.ts (hunk 2 above)
Comment (jayfgrollman) [reply modules/statics/src/coins.ts]: Done!

**Review by sachushaji — APPROVED** [review]

**Pull request merged**